front_end/core/host/RNPerfMetrics.ts

```diff
@@ -295,9 +352,24 @@ export type FuseboxSetClientMetadataFinishedEvent = Readonly<{
   }>,
 }>;
 
-export type ReactNativeChromeDevToolsEvent =
-    EntrypointLoadingStartedEvent|EntrypointLoadingFinishedEvent|DebuggerReadyEvent|BrowserVisibilityChangeEvent|
-    BrowserErrorEvent|RemoteDebuggingTerminatedEvent|DeveloperResourceLoadingStartedEvent|
-    DeveloperResourceLoadingFinishedEvent|FuseboxSetClientMetadataStartedEvent|FuseboxSetClientMetadataFinishedEvent;
+export type MemoryPanelActionStartedEvent = Readonly<{
+  eventName: 'MemoryPanelActionStarted',
+  params: Readonly<{
+    action: 'profiling' | 'sampling' | 'snapshot',
+  }>,
+}>;
+
+export type MemoryPanelActionFinishedEvent = Readonly<{
+  eventName: 'MemoryPanelActionFinished',
+  params: Readonly<{
+    action: 'profiling' | 'sampling' | 'snapshot',
+    success: boolean,
+  }>,
+}>;
+
+export type ReactNativeChromeDevToolsEvent = EntrypointLoadingStartedEvent|EntrypointLoadingFinishedEvent|
+    DebuggerReadyEvent|BrowserVisibilityChangeEvent|BrowserErrorEvent|RemoteDebuggingTerminatedEvent|
```

Comment:
Sidenote: I really wonder if we should adopt an `// @format` comment in our completely-forked files and have this enable Prettier instead 😅

Reply:
Yeah it's quite puzzling; would be good to learn why this is the existing preferred format